dubbo-common/src/main/java/com/alibaba/dubbo/common/utils/LRUCache.java

```diff
@@ -46,8 +46,8 @@ protected boolean removeEldestEntry(java.util.Map.Entry<K, V> eldest) {
 
     @Override
     public boolean containsKey(Object key) {
+        lock.lock();
```

Comment:
why move lock.lock() outside?

Reply:
If according to the original version, when lock.lock() reports an error, it will also enter the finally release lock. According to the Lock.unlock() document, an unchecked exception will be thrown when the non-lock holding thread calls this method: